Refresh.GameServer/Endpoints/Game/AnnouncementEndpoints.cs

```diff
@@ -16,57 +19,92 @@ namespace Refresh.GameServer.Endpoints.Game;
 
 public class AnnouncementEndpoints : EndpointGroup
 {
-    private static string AnnounceGetNotifications(GameDatabaseContext database, GameUser user, GameServerConfig config)
+    private static bool AnnounceGetNotifications(StringBuilder output, GameDatabaseContext database, GameUser user, GameServerConfig config)
     {
         List<GameNotification> notifications = database.GetNotificationsByUser(user, 5, 0).Items.ToList();
         int count = database.GetNotificationCountByUser(user);
-        if (count == 0) return string.Empty;
+        if (count == 0) return false;
 
         string s = count != 1 ? "s" : string.Empty;
 
-        string notificationText = $"Howdy, {user.Username}. You have {count} notification{s}:\n\n";
+        output.Append($"Howdy, {user.Username}. You have {count} notification{s}:\n\n");
         for (int i = 0; i < notifications.Count; i++)
         {
             GameNotification notification = notifications[i];
-            notificationText += $"  {notification.Title} ({i + 1}/{count}):\n" +
-                                $"    {notification.Text}\n\n";
+            output.Append($"  {notification.Title} ({i + 1}/{count}):\n");
+            output.Append($"    {notification.Text}\n\n");
         }
 
-        notificationText += $"To view more, or clear these notifications, you can visit the website at {config.WebExternalUrl}!\n";
-
-        return notificationText;
+        output.Append($"To view more, or clear these notifications, you can visit the website at {config.WebExternalUrl}!\n");
+        return true;
     }
 
-    private static string AnnounceGetAnnouncements(GameDatabaseContext database)
+    private static bool AnnounceGetAnnouncements(StringBuilder output, GameDatabaseContext database)
+    {
+        IEnumerable<GameAnnouncement> announcements = database.GetAnnouncements().ToList();
+        foreach (GameAnnouncement announcement in announcements)
+            output.Append($"{announcement.Title}: {announcement.Text}\n");
+        
+        return announcements.Any();
+    }
+    
+    private static bool AnnounceGetContest(StringBuilder output, Token token, GameDatabaseContext database, GameServerConfig config)
     {
-        IEnumerable<GameAnnouncement> announcements = database.GetAnnouncements();
-        // it's time to allocate
-        return announcements.Aggregate(string.Empty, (current, announcement) => current + $"{announcement.Title}: {announcement.Text}\n");
+        GameContest? contest = database.GetNewestActiveContest();
+        if (contest == null) return false;
+        
+        // only show contests for the current game
+        if (!contest.AllowedGames.Contains(token.TokenGame)) return false;
+        
+        output.Append("There's a contest live right now!\n\n");
+        output.AppendLine($"**{contest.ContestTitle}**");
```

Comment:
maybe its just me but i'd like if it had spaces
```suggestion
        output.AppendLine($"** {contest.ContestTitle} **");
```
\*\* Contest Name \*\*
vs
\*\*Contest Name\*\*